pkg/server/server.go

```diff
@@ -0,0 +1,285 @@
+package server
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"github.com/kcp-dev/kcp/pkg/etcd"
+	"github.com/kcp-dev/kcp/pkg/reconciler/apiresource"
+	"github.com/kcp-dev/kcp/pkg/reconciler/cluster"
+	clientv3 "go.etcd.io/etcd/client/v3"
+	"io/ioutil"
+	genericapiserver "k8s.io/apiserver/pkg/server"
+	"k8s.io/apiserver/pkg/storage/storagebackend"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	"k8s.io/kubernetes/pkg/genericcontrolplane"
+	"k8s.io/kubernetes/pkg/genericcontrolplane/clientutils"
+	"k8s.io/kubernetes/pkg/genericcontrolplane/options"
+	"net"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strconv"
+)
+
+type Server struct {
```

Comment:
What about adding comments on the public declarations in this file: `Server`, `Server.Run` and `NewServer` ?

Having a minimal example of how to start a server from a config would be great (even though it's quite obvious).

Reply:
I added clarifying comments with usage examples in the comments. Feel free to add or suggest any corrections to them.

Reply:
Thanks !

Reply:
However it seems there are some formatting problems in the example code of the `Server` documentation

Reply:
Nice !